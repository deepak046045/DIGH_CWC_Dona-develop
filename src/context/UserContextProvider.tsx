'use client';

import React, {
  Dispatch,
  ReactNode,
  SetStateAction,
  createContext,
  useMemo,
  useState,
} from 'react';
import {
  NavigationOption,
  ungatedNavigationOptions,
} from '@/models/NavigationOptions';
import { Appointment } from '@/models/Appointment';
import { AlertType } from '@/models/AlertType';
import mockAppointments from '@/mocks/mockAppointments';
import { CarePartner } from '@/models/CarePartner';
import { mockCarePartners } from '@/mocks/mockCarePartners';

type UserContextDataType = {
  isLoggedIn: boolean;
  setIsLoggedIn: Dispatch<SetStateAction<boolean>>;
  activeView: NavigationOption;
  setActiveView: Dispatch<NavigationOption>;
  alert: AlertType;
  setAlert: Dispatch<SetStateAction<AlertType>>;
  appointments: Appointment[];
  setAppointments: Dispatch<SetStateAction<Appointment[]>>;
  carePartners: CarePartner[];
  setCarePartners: Dispatch<SetStateAction<CarePartner[]>>;
};

export const UserContext = createContext<UserContextDataType>({
  isLoggedIn: false,
  setIsLoggedIn: () => {},
  activeView: ungatedNavigationOptions[0],
  setActiveView: () => {},
  alert: { isOpen: false, message: '', level: '' },
  setAlert: () => {},
  appointments: mockAppointments,
  setAppointments: () => {},
  carePartners: mockCarePartners,
  setCarePartners: () => {},
});

type Props = {
  children: ReactNode;
};

export function UserContextProvider({ children }: Props) {
  const [isLoggedIn, setIsLoggedIn] = useState<boolean>(false);
  const [activeView, setActiveView] = useState<NavigationOption>(
    ungatedNavigationOptions[0]
  );
  const [alert, setAlert] = useState<AlertType>({
    isOpen: false,
    message: '',
    level: '',
  });
  const [appointments, setAppointments] =
    useState<Appointment[]>(mockAppointments);
  const [carePartners, setCarePartners] =
    useState<CarePartner[]>(mockCarePartners);

  const values = useMemo(
    () => ({
      isLoggedIn,
      setIsLoggedIn,
      activeView,
      setActiveView,
      alert,
      setAlert,
      appointments,
      setAppointments,
      carePartners,
      setCarePartners,
    }),
    [isLoggedIn, activeView, alert, appointments, carePartners]
  );

  return <UserContext.Provider value={values}>{children}</UserContext.Provider>;
}
