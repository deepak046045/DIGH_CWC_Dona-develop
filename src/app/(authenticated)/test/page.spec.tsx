import { render } from '@testing-library/react';
import React from 'react';
import { UserContext } from '@/context/UserContextProvider';
import Home from '../page';
import '@testing-library/jest-dom';

jest.mock('crypto', () => ({ randomUUID: () => '123456789' }));
jest.mock('next/navigation', () => ({
  useRouter: jest.fn(() => ({ push: jest.fn() })),
}));

describe('Home Page', () => {
  it('should render page', () => {
    const { container } = render(<Home />);

    expect(container.querySelector('#alert_message')).not.toBeInTheDocument();
    expect(container).toMatchSnapshot();
  });

  it('should render page with alert', () => {
    const mockValue = {
      alert: { isOpen: true, level: 'info', message: 'this is an alert' },
    };

    const { container } = render(
      <UserContext.Provider value={mockValue}>
        <Home />
      </UserContext.Provider>
    );

    expect(container.querySelector('#alert_message')).toBeInTheDocument();
    expect(container).toMatchSnapshot();
  });
});
