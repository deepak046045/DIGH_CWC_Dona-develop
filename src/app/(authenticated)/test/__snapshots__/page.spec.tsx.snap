// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home Page should render page 1`] = `
<div>
  <div
    class="primary"
  />
  <div
    class="primary"
    id="test"
  />
  <div
    class="background"
  >
    <div
      class="welcomeSection gutter"
    >
      <div
        class="image_wrapper"
      >
        <img
          alt="ElderlyGroupSmiling"
          class="image"
          src="/images/elderly-group-smiling.png"
        />
      </div>
      <div
        class="welcomeSectionRightSide"
      >
        <div
          class="welcomeSectionHeaderText themeTypographyH1"
        >
          Welcome to Kisunla support.
        </div>
        <div
          class="welcomeSectionSubHeaderText"
        >
          Support resources for your treatment journey. Register to set up a
      calendar of appointments, set up reminders, and assign care
      partners.
        </div>
        <button
          class="lds-button primary"
          data-testid="sign-up-button"
          type="button"
        >
          Sign up now
        </button>
      </div>
    </div>
  </div>
  <div
    class="TreatmentIntro gutter"
  >
    <h2
      class="TreatmentIntro__title"
    >
      Recommended treatment
    </h2>
    <p
      class="TreatmentIntro__subtitle"
    >
      Setting you up for success in your treatment. Setting you up for success in your treatment.
    </p>
    <div
      class="TreatmentIntro__wrapper"
    >
      <p
        class="TreatmentIntro__text"
      >
        This treatment map serves as a suggested guide. Always follow your doctor’s directions. Insurance approval may be needed or the start of your infusion therapy may be delayed.
      </p>
    </div>
  </div>
  <div
    class="TreatmentMapList gutter"
    style="border-top: 1px solid black; border-bottom: 1px solid black; background: lightgrey; color: white; display: flex; align-content: center; justify-content: center; padding: 20px 0px;"
  >
    <p
      style="text-align: center; margin: 0px;"
    >
      Treatment Map List Placeholder
    </p>
  </div>
  <div
    class="TreatmentDownload gutter"
  >
    <p
      class="TreatmentDownload__text"
    >
      Always follow your doctor’s directions. Insurance approval may be needed or the start of your infusion therapy may be delayed.
    </p>
    <p
      class="TreatmentDownload__text"
    >
      For a printable version of the treatment map download the following.
      <button
        class="lds-button TreatmentDownload__map-download-btn"
        data-testid="download-button"
        type="button"
      >
        Treatment Map PDF
        <img
          alt="Download Icon"
          class="TreatmentDownload__map-download-btn-icon"
          data-nimg="1"
          decoding="async"
          height="19"
          loading="lazy"
          src="/icons/download-full-icon.svg"
          style="color: transparent;"
          width="19"
        />
      </button>
    </p>
    <div
      class="TreatmentDownload__divider"
    >
      <lds-hr />
    </div>
  </div>
  <div
    class="container gutter"
  >
    <div
      class="item"
    >
      <h2
        class="heading"
      >
        Treatment Touch Points
      </h2>
      <p
        class="details"
      >
        Throughout your treatment journey you’ll have regular touch points or visits with your health care provider. In the beginning of treatment these occur more frequently as you work to understand how effective the treatment is and set yourself up for success with ongoing treatment.
      </p>
    </div>
    <div
      class="item"
    >
      <h2
        class="heading"
      >
        Recommended Dosing
      </h2>
      <p
        class="details"
      >
        The recommended dosing shows a clear path for timing of doses, what kind of dose it is, and how much to take. For Kinsula your doses are in the form of infusions.
      </p>
    </div>
  </div>
  <div
    class="background"
  >
    <div
      class="customerSupportSignUpSection gutter"
    >
      <div
        class="customerSupportSignUpSectionHeader"
      >
        Let us assist you on your journey. Sign up on this website to
      </div>
      <div
        class="customerSupportSignUpSectionIconGroup"
      >
        <div
          class="customerSupportSignUpSectionIcon"
        >
          <img
            alt="CalendarIcon"
            data-nimg="1"
            decoding="async"
            height="60"
            loading="lazy"
            src="/icons/circle-calendar-icon.svg"
            style="color: transparent;"
            width="60"
          />
          <div
            class="customerSupportSignUpSectionText"
          >
            Personalize a calendar of your appointments
          </div>
        </div>
        <div
          class="customerSupportSignUpSectionIcon"
        >
          <img
            alt="CarePartnersIcon"
            data-nimg="1"
            decoding="async"
            height="60"
            loading="lazy"
            src="/icons/care-partners-icon.svg"
            style="color: transparent;"
            width="60"
          />
          <div
            class="customerSupportSignUpSectionText"
          >
            Assign care partners to assist you along your journey
          </div>
        </div>
        <div
          class="customerSupportSignUpSectionIcon"
        >
          <img
            alt="AppointmentRemindersIcon"
            data-nimg="1"
            decoding="async"
            height="60"
            loading="lazy"
            src="/icons/appointment-reminders-icon.svg"
            style="color: transparent;"
            width="60"
          />
          <div
            class="customerSupportSignUpSectionText"
          >
            Set up appointment reminders for you and your care partners
          </div>
        </div>
      </div>
      <button
        class="lds-button primary"
        data-testid="sign-up-button"
        type="button"
      >
        Sign up now
      </button>
    </div>
  </div>
  <div
    class="customerSupportSection gutter"
  >
    <div
      class="customerSupportSectionLeft"
    >
      <div
        class="customerSupportSectionLeftHeaderText"
      >
        Talk to a human
      </div>
      <div
        class="customerSupportHorizontalRule"
      >
        <lds-hr />
      </div>
      <div
        class="customerSupportSectionLeftSubheaderText"
      >
        If you have other questions about your medication, please contact us!
      </div>
    </div>
    <div
      class="customerSupportSectionRight"
    >
      <div
        class="customerSupportSectionPhone"
      >
        <img
          alt="PhoneIcon"
          data-nimg="1"
          decoding="async"
          height="37"
          loading="lazy"
          src="/icons/phone-icon.svg"
          style="color: transparent;"
          width="35"
        />
        <div
          class="customerSupportSectionPhoneText"
        >
          1-844-825-8966
        </div>
      </div>
      <div
        class="customerSupportSectionHours"
      >
        <img
          alt="HoursIcon"
          data-nimg="1"
          decoding="async"
          height="35"
          loading="lazy"
          src="/icons/purple-hours-icon.svg"
          style="color: transparent;"
          width="35"
        />
        <div
          class="customerSupportSectionHoursText"
        >
          Mon - Fri | 8:00 AM - 10:00 PM Eastern
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`Home Page should render page with alert 1`] = `
<div>
  <div
    class="primary"
  >
    <div
      aria-live="polite"
      class="lds-alert info"
      id="alert_message"
      role="alert"
    >
      <div
        class="lds-alert-message"
      >
        <button
          class="lds-button-base lds-alert-button"
        >
          <lds-icon
            class="close"
            inline="true"
            label="close"
            name="X"
          />
        </button>
        <svg
          aria-labelledby="lds-icon-InfoFill-title-123456789 lds-icon-InfoFill-description-123456789"
          class="lds-icon icon-InfoFill lds-alert-icon"
          focusable="false"
          role="img"
        >
          <title
            id="lds-icon-InfoFill-title-123456789"
          >
            Info
          </title>
          <desc
            id="lds-icon-InfoFill-description-123456789"
          />
          <svg
            viewBox="0 0 256 256"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M128,24A104,104,0,1,0,232,128,104.11,104.11,0,0,0,128,24Zm-4,48a12,12,0,1,1-12,12A12,12,0,0,1,124,72Zm12,112a16,16,0,0,1-16-16V128a8,8,0,0,1,0-16,16,16,0,0,1,16,16v40a8,8,0,0,1,0,16Z"
            />
          </svg>
        </svg>
        <div
          class="message"
        >
          this is an alert
        </div>
      </div>
    </div>
  </div>
  <div
    class="primary"
    id="test"
  />
  <div
    class="background"
  >
    <div
      class="welcomeSection gutter"
    >
      <div
        class="image_wrapper"
      >
        <img
          alt="ElderlyGroupSmiling"
          class="image"
          src="/images/elderly-group-smiling.png"
        />
      </div>
      <div
        class="welcomeSectionRightSide"
      >
        <div
          class="welcomeSectionHeaderText themeTypographyH1"
        >
          Welcome to Kisunla support.
        </div>
        <div
          class="welcomeSectionSubHeaderText"
        >
          Support resources for your treatment journey. Register to set up a
      calendar of appointments, set up reminders, and assign care
      partners.
        </div>
        <button
          class="lds-button primary"
          data-testid="sign-up-button"
          type="button"
        >
          Sign up now
        </button>
      </div>
    </div>
  </div>
  <div
    class="TreatmentIntro gutter"
  >
    <h2
      class="TreatmentIntro__title"
    >
      Recommended treatment
    </h2>
    <p
      class="TreatmentIntro__subtitle"
    >
      Setting you up for success in your treatment. Setting you up for success in your treatment.
    </p>
    <div
      class="TreatmentIntro__wrapper"
    >
      <p
        class="TreatmentIntro__text"
      >
        This treatment map serves as a suggested guide. Always follow your doctor’s directions. Insurance approval may be needed or the start of your infusion therapy may be delayed.
      </p>
    </div>
  </div>
  <div
    class="TreatmentMapList gutter"
    style="border-top: 1px solid black; border-bottom: 1px solid black; background: lightgrey; color: white; display: flex; align-content: center; justify-content: center; padding: 20px 0px;"
  >
    <p
      style="text-align: center; margin: 0px;"
    >
      Treatment Map List Placeholder
    </p>
  </div>
  <div
    class="TreatmentDownload gutter"
  >
    <p
      class="TreatmentDownload__text"
    >
      Always follow your doctor’s directions. Insurance approval may be needed or the start of your infusion therapy may be delayed.
    </p>
    <p
      class="TreatmentDownload__text"
    >
      For a printable version of the treatment map download the following.
      <button
        class="lds-button TreatmentDownload__map-download-btn"
        data-testid="download-button"
        type="button"
      >
        Treatment Map PDF
        <img
          alt="Download Icon"
          class="TreatmentDownload__map-download-btn-icon"
          data-nimg="1"
          decoding="async"
          height="19"
          loading="lazy"
          src="/icons/download-full-icon.svg"
          style="color: transparent;"
          width="19"
        />
      </button>
    </p>
    <div
      class="TreatmentDownload__divider"
    >
      <lds-hr />
    </div>
  </div>
  <div
    class="container gutter"
  >
    <div
      class="item"
    >
      <h2
        class="heading"
      >
        Treatment Touch Points
      </h2>
      <p
        class="details"
      >
        Throughout your treatment journey you’ll have regular touch points or visits with your health care provider. In the beginning of treatment these occur more frequently as you work to understand how effective the treatment is and set yourself up for success with ongoing treatment.
      </p>
    </div>
    <div
      class="item"
    >
      <h2
        class="heading"
      >
        Recommended Dosing
      </h2>
      <p
        class="details"
      >
        The recommended dosing shows a clear path for timing of doses, what kind of dose it is, and how much to take. For Kinsula your doses are in the form of infusions.
      </p>
    </div>
  </div>
  <div
    class="background"
  >
    <div
      class="customerSupportSignUpSection gutter"
    >
      <div
        class="customerSupportSignUpSectionHeader"
      >
        Let us assist you on your journey. Sign up on this website to
      </div>
      <div
        class="customerSupportSignUpSectionIconGroup"
      >
        <div
          class="customerSupportSignUpSectionIcon"
        >
          <img
            alt="CalendarIcon"
            data-nimg="1"
            decoding="async"
            height="60"
            loading="lazy"
            src="/icons/circle-calendar-icon.svg"
            style="color: transparent;"
            width="60"
          />
          <div
            class="customerSupportSignUpSectionText"
          >
            Personalize a calendar of your appointments
          </div>
        </div>
        <div
          class="customerSupportSignUpSectionIcon"
        >
          <img
            alt="CarePartnersIcon"
            data-nimg="1"
            decoding="async"
            height="60"
            loading="lazy"
            src="/icons/care-partners-icon.svg"
            style="color: transparent;"
            width="60"
          />
          <div
            class="customerSupportSignUpSectionText"
          >
            Assign care partners to assist you along your journey
          </div>
        </div>
        <div
          class="customerSupportSignUpSectionIcon"
        >
          <img
            alt="AppointmentRemindersIcon"
            data-nimg="1"
            decoding="async"
            height="60"
            loading="lazy"
            src="/icons/appointment-reminders-icon.svg"
            style="color: transparent;"
            width="60"
          />
          <div
            class="customerSupportSignUpSectionText"
          >
            Set up appointment reminders for you and your care partners
          </div>
        </div>
      </div>
      <button
        class="lds-button primary"
        data-testid="sign-up-button"
        type="button"
      >
        Sign up now
      </button>
    </div>
  </div>
  <div
    class="customerSupportSection gutter"
  >
    <div
      class="customerSupportSectionLeft"
    >
      <div
        class="customerSupportSectionLeftHeaderText"
      >
        Talk to a human
      </div>
      <div
        class="customerSupportHorizontalRule"
      >
        <lds-hr />
      </div>
      <div
        class="customerSupportSectionLeftSubheaderText"
      >
        If you have other questions about your medication, please contact us!
      </div>
    </div>
    <div
      class="customerSupportSectionRight"
    >
      <div
        class="customerSupportSectionPhone"
      >
        <img
          alt="PhoneIcon"
          data-nimg="1"
          decoding="async"
          height="37"
          loading="lazy"
          src="/icons/phone-icon.svg"
          style="color: transparent;"
          width="35"
        />
        <div
          class="customerSupportSectionPhoneText"
        >
          1-844-825-8966
        </div>
      </div>
      <div
        class="customerSupportSectionHours"
      >
        <img
          alt="HoursIcon"
          data-nimg="1"
          decoding="async"
          height="35"
          loading="lazy"
          src="/icons/purple-hours-icon.svg"
          style="color: transparent;"
          width="35"
        />
        <div
          class="customerSupportSectionHoursText"
        >
          Mon - Fri | 8:00 AM - 10:00 PM Eastern
        </div>
      </div>
    </div>
  </div>
</div>
`;
